Makefile

```diff
@@ -1,12 +1,42 @@
 current_dir := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+CARGO ?= $(shell which cargo)
+
+ifdef RUST_TARGET
+	TARGET = --target ${RUST_TARGET}
+endif
 
 build:
-	cargo build --features="fusedev"
-	cargo build --features="virtiofs"
-	cargo build --features="vhost-user-fs"
-	cargo build --features="fusedev,async-io"
-	cargo build --features="virtiofs,async-io"
-	cargo build --features="vhost-user-fs,async-io"
+	${CARGO} build ${TARGET} --features="fusedev"
+	${CARGO} build ${TARGET} --features="virtiofs"
+	${CARGO} build ${TARGET} --features="vhost-user-fs"
+	${CARGO} build ${TARGET} --features="fusedev,async-io"
+	${CARGO} build ${TARGET} --features="virtiofs,async-io"
+	${CARGO} build ${TARGET} --features="vhost-user-fs,async-io"
+
+check: build
+	${CARGO} fmt -- --check
+	${CARGO} clippy ${TARGET} --features="fusedev" --no-default-features -- -Dwarnings
+	${CARGO} clippy ${TARGET} --features="virtiofs" --no-default-features -- -Dwarnings
+	${CARGO} clippy ${TARGET} --features="vhost-user-fs" --no-default-features -- -Dwarnings
+	${CARGO} clippy ${TARGET} --features="fusedev,virtiofs" --no-default-features -- -Dwarnings
+
+test:
+	cargo test ${TARGET} --features="fusedev" --no-default-features -- --nocapture --skip integration
```

Comment:
Should the "cargo" cmd in test be replaced with ${CARGO} as well?

Reply:
Nop, we do not want to use cross to run the test as it would launch a docker container which: 1. slows things down, and 2. it cannot reallly do cross-architecture verification.